benchmarks/test_sharded_parity.py

```diff
@@ -0,0 +1,320 @@
+def test_ddp_sharded_plugin_correctness_one_device():
+    # Allow slightly slower speed due to one CPU doing additional sequential memory saving calls
+    plugin_parity_test(
+        accelerator='ddp_cpu',
+        max_percent_speed_diff=0.5,
+        plugin=DDPShardedPlugin(),
+        model_cls=SeedTrainLoaderModel
+    )
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="requires GPU machine")
+@pytest.mark.skipif(platform.system() == "Windows",
+                    reason="Distributed training is not supported on Windows")
+@pytest.mark.skipif(not FAIRSCALE_AVAILABLE, reason="Fairscale is not available")
+def test_ddp_sharded_plugin_correctness_one_gpu():
+    plugin_parity_test(
+        gpus=1,
+        accelerator='ddp_spawn',
+        plugin=DDPShardedPlugin(),
+        model_cls=SeedTrainLoaderModel
+    )
+
+
+@pytest.mark.skipif(not NATIVE_AMP_AVAILABLE, reason="Requires native AMP")
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="requires GPU machine")
+@pytest.mark.skipif(platform.system() == "Windows",
+                    reason="Distributed training is not supported on Windows")
+@pytest.mark.skipif(not FAIRSCALE_AVAILABLE, reason="Fairscale is not available")
+def test_ddp_sharded_plugin_correctness_amp_one_gpu():
+    plugin_parity_test(
+        gpus=1,
+        precision=16,
+        accelerator='ddp_spawn',
+        plugin=DDPShardedPlugin(),
+        model_cls=SeedTrainLoaderModel
+    )
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+@pytest.mark.skipif(platform.system() == "Windows",
+                    reason="Distributed training is not supported on Windows")
+@pytest.mark.skipif(not FAIRSCALE_AVAILABLE, reason="Fairscale is not available")
+def test_ddp_sharded_plugin_correctness_multi_gpu():
+    plugin_parity_test(
+        gpus=2,
+        accelerator='ddp_spawn',
+        plugin=DDPShardedPlugin(),
+        model_cls=SeedTrainLoaderModel
+    )
+
+
+@pytest.mark.skipif(not NATIVE_AMP_AVAILABLE, reason="Requires native AMP")
+@pytest.mark.skipif(platform.system() == "Windows",
+                    reason="Distributed training is not supported on Windows")
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+@pytest.mark.skipif(not FAIRSCALE_AVAILABLE, reason="Fairscale is not available")
+def test_ddp_sharded_plugin_correctness_amp_multi_gpu():
+    plugin_parity_test(
+        gpus=2,
+        precision=16,
+        accelerator='ddp_spawn',
+        plugin=DDPShardedPlugin(),
+        model_cls=SeedTrainLoaderModel
+    )
+
+
+@pytest.mark.skipif(not FAIRSCALE_AVAILABLE, reason="Fairscale is not available")
+@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+@pytest.mark.skipif(not os.getenv("PL_RUNNING_SPECIAL_TESTS", '0') == '1',
+                    reason="test should be run outside of pytest")
```

Comment:
how this `os.getenv("PL_RUNNING_SPECIAL_TESTS", '0')` works if you patch it above with `@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})`?